rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -185,22 +192,42 @@ DDS_TypeCode * create_type_code(
           std::string field_type_name = std::string(sub_members->package_name_) + "::msg::dds_::" +
             sub_members->message_name_ + "_";
           member_type_code = create_type_code(field_type_name, sub_members, participant_qos);
+          if (!member_type_code) {
+            // error string was set within the function
+            return NULL;
+          }
         }
         break;
       default:
-        printf("unknown type id %u\n", member->type_id_);
-        throw std::runtime_error("unknown type");
+        throw std::runtime_error(
+          std::string("unknown type id ") + std::to_string(member->type_id_));
```

Comment:
Currently all `default` cases throw an exception. It is similar to other code where we do the same when an *enum* is not in the range we expect it to be. Not sure if we want to keep it that way.